UBCFormulaElectric/Consolidated-Firmware
# Auto-generate code to transmit periodic CAN messages

### Summary

Automate the process of sending periodic CAN messages by leveraging the sender and cycle time information inside DBC.
 
### Changelist 

- Remove .sym and replaced it with .dbc
- Auto-generate code to transmit periodic CAN messages. This does *not* cover any non-periodic CAN messages like motor shutdown. 

### Testing Done

- Manually performed sanity check between .dbc and the generated code
- Integrated the auto-generated code on FSM and DCM, and observe that the appropriate periodic CAN messages are being transmitted with the expected DLC and data.
- Toggle a CAN signal in 1Hz task and observer that the appropriate changes is broadcasted. For example, I toggled `papps_out_range` in the 1Hz task like so:
```
App_CanMsgsTx_GetCanTxPayloads()->fsm_errors.papps_out_of_range = 1 ^ App_CanMsgsTx_GetCanTxPayloads()->fsm_errors.papps_out_of_range;
```
And I observed that the bit is alternating in the CAN output
```
 (000.100005)  can0  040   [8]  00 00 00 00 00 00 00 00 :: FSM_ERRORS(papps_out_of_range: 0, sapps_out_of_range: 0)
 (000.199945)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.299737)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.299957)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.200126)  can0  040   [8]  01 00 00 00 00 00 00 00 :: FSM_ERRORS(papps_out_of_range: 1, sapps_out_of_range: 0)
 (000.099861)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.299903)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.300004)  can0  041   [0]  :: FSM_HEARTBEAT()
 (000.300157)  can0  040   [8]  00 00 00 00 00 00 00 00 :: FSM_ERRORS(papps_out_of_range: 0, sapps_out_of_range: 0)
```
- Ran the code for 10 minutes without observing any weird bugs like system reboots.
- Make sure the CAN autogen code is only re-generated when .dbc is modified

### Resolved Issues


### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code standards detailed in http://svformulaep1.teams.apsc.ubc.ca:8090/display/TR/Workflow (*This will save time for both you and the reviewer!*).
